cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java

```diff
@@ -779,6 +781,72 @@ public void testBuildEmptyPackageWithCompilerPlugin() throws IOException {
                 getOutput("build-empty-package.txt"));
     }
 
+    @Test(description = "Build a ballerina project with the flag dump-graph")
+    public void testBuildBalProjectWithDumpGraphFlag() throws IOException {
+        Path dumpGraphResourcePath = this.testResources.resolve("projectsForDumpGraph");
+        BCompileUtil.compileAndCacheBala(dumpGraphResourcePath.resolve("package_c"), testDistCacheDirectory,
+                projectEnvironmentBuilder);
+        BCompileUtil.compileAndCacheBala(dumpGraphResourcePath.resolve("package_b"), testDistCacheDirectory,
+                projectEnvironmentBuilder);
+
+        Path projectPath = dumpGraphResourcePath.resolve("package_a");
+        System.setProperty("user.dir", projectPath.toString());
+
+        java.io.ByteArrayOutputStream out = new java.io.ByteArrayOutputStream();
+        System.setOut(new java.io.PrintStream(out));
+
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+        new CommandLine(buildCommand).parseArgs("--dump-graph");
+        buildCommand.execute();
+        String buildLog = readOutput(true).replaceAll("\r", "");
+        String dependencyGraphLog = out.toString().replaceAll("\r", "");
+
+        Assert.assertTrue(dependencyGraphLog.contains(getOutput("dump-graph-output.txt")));
+        Assert.assertTrue(buildLog.contains(getOutput("build-bal-project-with-dump-graph.txt")));
+        Assert.assertTrue(projectPath.resolve("target").resolve("cache").resolve("foo")
+                .resolve("package_a").resolve("0.1.0").resolve("java11")
+                .resolve("foo-package_a-0.1.0.jar").toFile().exists());
+
+        try {
+            Files.walk(projectPath.resolve("target")).sorted(Comparator.reverseOrder()).map(Path::toFile)
```

Comment:
Are you trying to delete the target dir in this case?

I think you can make use of `ProjectUtils.deleteDirectory(<path>)`. I think its used in `testBuildBalFileWithOutputFlag` test. If it works could you update this section